src/pr_merge.rs

```diff
@@ -14,131 +14,104 @@ use messenger;
-            return Err(format!("PR branch already exists on origin: '{}'", pr_branch_name).into());
-        }
-
-        let (title, body) = self.cherry_pick(
-            &git,
-            &merge_commit_sha,
-            &pr_branch_name,
-            pull_request.number,
-            &target_branch,
-            &pull_request.base.ref_name,
-        )?;
-
-        git.run(&["push", "origin", &format!("{}:{}", pr_branch_name, pr_branch_name)])?;
-
-        let new_pr = self.session.create_pull_request(
-            owner,
-            repo,
-            &title,
-            &body,
-            &pr_branch_name,
-            &target_branch,
-        )?;
-
-        let assignees: Vec<String> = pull_request.assignees.iter().map(|a| a.login().to_string()).collect();
-        self.session.assign_pull_request(owner, repo, new_pr.number, assignees)?;
-
-        Ok(new_pr)
+    let merge_commit_sha;
+    if let Some(ref sha) = pull_request.merge_commit_sha {
+        merge_commit_sha = sha;
+    } else {
+        return Err(format!("Pull Request #{} has no merge commit.", pull_request.number).into());
     }
 
-    fn cherry_pick(
-        &self,
-        git: &Git,
-        commit_hash: &str,
-        pr_branch_name: &str,
-        pr_number: u32,
-        target_branch: &str,
-        orig_base_branch: &str,
-    ) -> Result<(String, String)> {
-        git.checkout_branch(pr_branch_name, &format!("origin/{}", target_branch))?;
-
-        // cherry-pick!
-        git.run(
-            &[
-                "-c",
-                "merge.renameLimit=999999",
-                "cherry-pick",
-                "--allow-empty",
-                "-X",
-                "ignore-all-space",
-                commit_hash,
-            ],
-        )?;
-
-        let desc = git.get_commit_desc(commit_hash)?;
-        let desc = make_merge_desc(desc, commit_hash, pr_number, target_branch, orig_base_branch);
-
-        // change commit message
-        git.run_with_stdin(
-            &["commit", "--amend", "-F", "-"],
-            &format!("{}\n\n{}", &desc.0, &desc.1),
-        )?;
-
-        Ok(desc)
+    // strip everything before last slash
+    let regex = Regex::new(r".*/").unwrap();
+    let pr_branch_name =
+        format!("{}-{}", regex.replace(&pull_request.head.ref_name, ""), regex.replace(&target_branch, ""));
+
+    // make sure there isn't already such a branch
+    let current_remotes = git.run(&["ls-remote", "--heads"])?;
+    if current_remotes.contains(&format!("refs/heads/{}", pr_branch_name)) {
+        return Err(format!("PR branch already exists on origin: '{}'", pr_branch_name).into());
     }
+
+    let (title, body) = cherry_pick(
+        &git,
+        &merge_commit_sha,
+        &pr_branch_name,
+        pull_request.number,
+        &target_branch,
+        &pull_request.base.ref_name,
+    )?;
+
+    git.run(&["push", "origin", &format!("{}:{}", pr_branch_name, pr_branch_name)])?;
```

Comment:
`&format!("{}:{}", pr_branch_name, pr_branch_name)]` - i've done this typically for renaming the branch but in this case it's the same.  that's not necessarrily an issue but did you mean to do it this way?  It made me think perhaps you meant to rename it.

Reply:
I believe this was to be totally explicit and avoid assumptions about what the git push.default config setting. Could probably have done `format!("HEAD:{}", pr_branch_name)` to accomplish this without confusion.

Reply:
(note: if you ignore whitespace on this PR, the diff gets a lot smaller... This line is not actually a change)